### PR TITLE
[IA-2663] Increase k8s QPS and Burst client configs

### PR DESCRIPTION
**Pull request by rtitle**
https://broadworkbench.atlassian.net/browse/IA-2663

Hoping this helps with the `Throttling request` errors. Also bumped helm and k8s go package versions.

Tested in a fiab, still need to test on perf.

### CODE REVIEW

File: helm-go-lib/go.mod [hunk 1]
```diff
@@ -1,32 +1,15 @@
-module helm.sh/helm/v3@v3.2.4
+module broadinstitute/helm-scala-sdk
```
Comment (rtitle) [inline helm-go-lib/go.mod:1]: I think this is supposed to be _our_ module name

File: helm-go-lib/go.mod [hunk 2]
```diff
@@ -1,32 +1,15 @@
...
 require (
-	9fans.net/go v0.0.2 // indirect
-	github.com/Masterminds/semver v1.5.0 // indirect
-	github.com/helm/helm v2.16.6+incompatible // indirect
-	gonum.org/v1/netlib v0.0.0-20190331212654-76723241ea4e // indirect
-	helm.sh/helm/v3 v3.2.0 // indirect
-	k8s.io/helm v2.16.6+incompatible // indirect
-	sigs.k8s.io/structured-merge-diff v1.0.1-0.20191108220359-b1b620dd3f06 // indirect
+	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
+	helm.sh/helm/v3 v3.5.3
```
Comment (rtitle) [inline helm-go-lib/go.mod:12]: Bumped `helm` and `k8s` to the latest. Also removed some dependencies which aren't needed.

File: helm-go-lib/go.sum [hunk 3]
```diff
@@ -1,257 +1,297 @@
-9fans.net/go v0.0.2 h1:RYM6lWITV8oADrwLfdzxmt8ucfW6UtP9v1jg4qAbqts=
```
Comment (rtitle) [inline helm-go-lib/go.sum:1]: This file is aut-generated

File: helm-go-lib/main.go [hunk 4]
```diff
@@ -181,10 +182,30 @@ func buildActionConfig(namespace string, kubeToken string, apiServer string, caF
...
-	if err := actionConfig.Init(kubeConfig, namespace, os.Getenv("HELM_DRIVER"), log.Printf); err != nil {
+	if err := actionConfig.Init(customConfig, namespace, os.Getenv("HELM_DRIVER"), log.Printf); err != nil {
 		return nil, err
 	}
 
 	return actionConfig, nil
 }
+
+// Override ToRESTConfig to set QPS and Burst
+type CustomConfigFlags struct {
```
Comment (rtitle) [inline helm-go-lib/main.go:196]: This is the crux: I am overridding the `ConfigFlags` struct (which implements the `RESTClientGetter` interface) to be able to set `QPS`/`Burst` on the `RESTConfig`. On a fiab I see the log statement, so I think it's working. (Let me know if there's a more Go-idiomatic way to do this.)

**Review by Qi77Qi — COMMENTED** [review]
think we need to bump the library version?

**Comment by rtitle** [comment]
Oh yeah - probably a good idea

File: helm-go-lib/main.go [hunk 5]
```diff
@@ -181,10 +182,32 @@ func buildActionConfig(namespace string, kubeToken string, apiServer string, caF
...
 	actionConfig = new(action.Configuration)
-	if err := actionConfig.Init(kubeConfig, namespace, os.Getenv("HELM_DRIVER"), log.Printf); err != nil {
+	if err := actionConfig.Init(customConfig, namespace, os.Getenv("HELM_DRIVER"), log.Printf); err != nil {
 		return nil, err
 	}
 
 	return actionConfig, nil
 }
+
+// Override ConfigFlags struct (which implments the RESTClientGetter interface)
```
Comment (rtitle) [inline helm-go-lib/main.go:195]: This is the crux of the fix here

**Review by Qi77Qi — APPROVED** [review]
👍 

can you create a release note similar to https://github.com/broadinstitute/helm-scala-sdk/releases/tag/0.0.2

**Review by kyuksel — COMMENTED** [review]
🚀

**Pull request merged**